### PR TITLE
use configurable logger instead of slf4j logger everywhere.

### PR DESCRIPTION
**Pull request by moderakh**
- use configurable logger instead of slf4j logger everywhere in the spark connector layer (doesn't touch the SDK).
- wired up per request operation/response (for query, and write operations) against spark diagnostics.

please look at these classes for starting to review:
DiagnosticsProvider.scala
SparkTaskContext.scala
PointWriter

For point writes when diagnostics enabled we will get something as follows:

```
21/06/01 11:18:21 INFO PointWriter: SparkTaskContext(correlationActivityId=abc7c1bc-1d3d-456b-bd43-27f63419f3c8,stageId=0,partitionId=0,details=), response: response{statusCode:200, headers{activityId:befce693-c305-11eb-a373-b51dba521584}}
21/06/01 11:18:21 INFO PointWriter: UpsertOperation(SparkTaskContext(correlationActivityId=abc7c1bc-1d3d-456b-bd43-27f63419f3c8,stageId=0,partitionId=0,details=PointWriter),CosmosItemIdentifier(bat,["bat"])) completed
21/06/01 11:18:21 INFO PointWriter: SparkTaskContext(correlationActivityId=abc7c1bc-1d3d-456b-bd43-27f63419f3c8,stageId=0,partitionId=0,details=), response: response{statusCode:200, headers{activityId:befd82d4-c305-11eb-a373-b51dba521584}}
21/06/01 11:18:21 INFO PointWriter: UpsertOperation(SparkTaskContext(correlationActivityId=abc7c1bc-1d3d-456b-bd43-27f63419f3c8,stageId=0,partitionId=0,details=PointWriter),CosmosItemIdentifier(mouse,["mouse"])) completed
21/06/01 11:18:21 INFO PointWriter: SparkTaskContext(correlationActivityId=abc7c1bc-1d3d-456b-bd43-27f63419f3c8,stageId=0,partitionId=0,details=), response: response{statusCode:200, headers{activityId:befda9e5-c305-11eb-a373-b51dba521584}}
21/06/01 11:18:21 INFO PointWriter: UpsertOperation(SparkTaskContext(correlationActivityId=abc7c1bc-1d3d-456b-bd43-27f63419f3c8,stageId=0,partitionId=0,details=PointWriter),CosmosItemIdentifier(horse,["horse"])) completed
```

for bulk write when diagnostics enabled we get something like following.

```
21/06/01 11:13:59 INFO BulkWriter: SparkTaskContext(correlationActivityId=1fe6232b-809c-4e2b-98eb-c10708656300,stageId=0,partitionId=0,details=), request: request{operationType:Batch, resourceType:Document, partitionKey:null, resourceAddress:dbs/testDB/colls/testContainer, headers{activityId:22f471f7-c305-11eb-98b7-f93f10f7d1fe, items:3}}
21/06/01 11:13:59 INFO BulkWriter: SparkTaskContext(correlationActivityId=1fe6232b-809c-4e2b-98eb-c10708656300,stageId=0,partitionId=0,details=), response: response{statusCode:200, headers{activityId:22f471f7-c305-11eb-98b7-f93f10f7d1fe, itemCount:3, continuationToken:null}}
```
similar to point write, if we agree on the design I can add the per item details to bulk as

### CODE REVIEW

File: sdk/cosmos/azure-cosmos-spark_3-1_2-12/src/main/scala/com/azure/cosmos/spark/diagnostics/CosmosItemIdentifier.scala [hunk 1]
```diff
@@ -0,0 +1,8 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos.spark.diagnostics
+
+import com.azure.cosmos.models.PartitionKey
+
+case class CosmosItemIdentifier(itemId: String, partitionKeyValue: PartitionKey)
```
Comment (FabianMeiswinkel) [inline sdk/cosmos/azure-cosmos-spark_3-1_2-12/src/main/scala/com/azure/cosmos/spark/diagnostics/CosmosItemIdentifier.scala:8]: I think we should add (at least otionally) eTag here?

File: sdk/cosmos/azure-cosmos-spark_3-1_2-12/src/main/scala/org/apache/spark/source/MySource.scala [hunk 2]
```diff
@@ -0,0 +1,16 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package org.apache.spark.source
```
Comment (FabianMeiswinkel) [inline sdk/cosmos/azure-cosmos-spark_3-1_2-12/src/main/scala/org/apache/spark/source/MySource.scala:4]: I assume this file should not be checked-in yet?

**Review by FabianMeiswinkel — APPROVED** [review]
Thanks Mo - I like the approach!

File: sdk/cosmos/azure-cosmos-spark_3-1_2-12/src/main/scala/org/apache/spark/source/MySource.scala (hunk 2 above)
Comment (moderakh) [reply sdk/cosmos/azure-cosmos-spark_3-1_2-12/src/main/scala/org/apache/spark/source/MySource.scala]: thanks for catching this. Yes some local work. It shouldn't get checked in.

I removed it.

File: sdk/cosmos/azure-cosmos-spark_3-1_2-12/src/main/scala/com/azure/cosmos/spark/diagnostics/CosmosItemIdentifier.scala (hunk 1 above)
Comment (moderakh) [reply sdk/cosmos/azure-cosmos-spark_3-1_2-12/src/main/scala/com/azure/cosmos/spark/diagnostics/CosmosItemIdentifier.scala:8]: sure. thanks for the comment. I am tracking this in an upcoming PR.

**Pull request merged**